src/main/java/com/blamejared/crafttweaker/impl/predicate/TriState.java

```diff
@@ -0,0 +1,24 @@
+package com.blamejared.crafttweaker.impl.predicate;
+
+import com.blamejared.crafttweaker.api.annotations.ZenRegister;
+import com.blamejared.crafttweaker_annotations.annotations.Document;
+import org.openzen.zencode.java.ZenCodeType;
+
+@ZenRegister()
+@ZenCodeType.Name("crafttweaker.api.predicate.TriState")
+@Document("vanilla/api/predicate/TriState")
+public enum TriState {
+    TRUE(true),
+    FALSE(false),
+    UNSET(null);
+
+    private final Boolean bool;
+
+    TriState(final Boolean bool) {
+        this.bool = bool;
+    }
+
+    public Boolean toBoolean() {
+        return this.bool;
+    }
```

Comment:
Repeat of the idea stated earlier:
Add a method here 
```java
public boolean isSameOrUnset(boolean value) {
    return this == UNSET || value == bool.booleanValue();
}
```

May make some conditional checks easier in the other classes when you don't have to explicitly check for unset 😉